### ISSUE TITLE
Unable to Update Containers - i/o Timeout via DNS Lookup?

### ISSUE THREAD

**Issue by phikai** [containrrr/watchtower#192]
I'm getting the following error on a container update:

```
time="2018-05-17T23:48:48-05:00" level=info msg="Unable to update container /portainer, err='Error response from daemon: Get https://registry-1.docker.io/v2/portainer/portainer/manifests/latest: Get https://auth.docker.io/token?scope=repository%3Aportainer%2Fportainer%3Apull&service=registry.docker.io: dial tcp: lookup auth.docker.io on xxx.xxx.xxx.xxx:53: read udp xxx.xxx.xxx.xxx:45264->xxx.xxx.xxx.xxx:53: i/o timeout'. Proceeding to next." 
```

Docker-Compose to launch Watchtower is:

```
# ----------------------------------------
    # WATCHTOWER
    # Automatic Update Service for Docker VMs
    # ----------------------------------------
    watchtower:
        image: v2tec/watchtower
        container_name: watchtower
        restart: unless-stopped
        environment:
            - TZ=${TZ}
            - WATCHTOWER_NOTIFICATIONS=email
            - WATCHTOWER_NOTIFICATION_EMAIL_FROM=${SMTP_FROM}
            - WATCHTOWER_NOTIFICATION_EMAIL_TO=${WATCHTOWER_EMAIL}
            - WATCHTOWER_NOTIFICATION_EMAIL_SERVER=${SMTP_SERVER}
            - WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PORT=${SMTP_PORT}
            - WATCHTOWER_NOTIFICATION_EMAIL_SERVER_USER=${SMTP_USER}
            - WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PASSWORD=${SMTP_PASS}
        network_mode: 'bridge'
        volumes:
            - '/var/run/docker.sock:/var/run/docker.sock'
            - '/etc/ssl/certs:/etc/ssl/certs'
        command: portainer --schedule @midnight --cleanup
```

Potentially the only thing I can see here is that I run Pi-Hole in front of my DNS, but I don't see anything being blocked there...

**Comment by ssbarnea** [comment]
I think it may be related to https://github.com/v2tec/watchtower/issues/203

**Comment by ovizii** [comment]
I'm having similar problems (seeing the exact same problem with portainer and other containers too). I just copied the URL from your error message above and did a curl -I from the machine with the issue:

```
curl -I https://registry-1.docker.io/v2/portainer/portainer/manifests/latest
HTTP/1.1 401 Unauthorized
Content-Type: application/json; charset=utf-8
Docker-Distribution-Api-Version: registry/2.0
Www-Authenticate: Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:portainer/portainer:pull"
Date: Tue, 31 Jul 2018 10:38:00 GMT
Content-Length: 162
Strict-Transport-Security: max-age=31536000

```

Does this mean we need to supply credentials?

**Comment by simonk83** [comment]
Same issue here, I think.   Every single container fails with the following:

```Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)```

I'm also running Pihole for reference, but I've specified 8.8.8.8 in resolv.conf (Debian 9) to try and rule that out (didn't help).

Watchtower attempting to update itself gave this:

```Unable to update container /watchtower, err='Error response from daemon: Get https://registry-1.docker.io/v2/v2tec/watchtower/manifests/latest: Get https://auth.docker.io/token?scope=repository%3Av2tec%2Fwatchtower%3Apull&service=registry.docker.io: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)'.```

**Comment by Aerya** [comment]
Hi, same here. Any update/ETA ?

**Comment by ovizii** [comment]
I'm getting new (similar) errors with all containers for the last 2 days:

`2018-11-25 22:57:38 (info): Unable to update container /unifi, err='Error response from daemon: error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n"'. Proceeding to next.`

`2018-11-25 23:27:41 (info): Unable to update container /portainer, err='Error response from daemon: error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n"'. Proceeding to next.`

**Comment by waja** [comment]
> I'm getting new (similar) errors with all containers for the last 2 days:
> 
> `2018-11-25 22:57:38 (info): Unable to update container /unifi, err='Error response from daemon: error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n"'. Proceeding to next.`

This is related to #244 or better https://github.com/docker/hub-feedback/issues/1693.

**Comment by ovizii** [comment]
Thanks for pointing me in the right direction. I have now subscribed to those issues.

**Comment by simonaronsson** [comment]
This is solved according to #244. Closing issue.
